stomp_plugins/src/noise_generators/cartesian_constraints_sampling.cpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file cartesian_constraints_sampling.cpp
+ * @brief This a normal distribution noisy trajectory update generator in cartesian space
+ *
+ * @author Michal Dobis
+ * @date June 6, 2021
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2021, Photoneo
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <stomp_plugins/noise_generators/cartesian_constraints_sampling.h>
+#include <stomp_moveit/utils/multivariate_gaussian.h>
+#include <XmlRpcException.h>
+#include <pluginlib/class_list_macros.h>
+#include <ros/console.h>
+
+PLUGINLIB_EXPORT_CLASS(stomp_moveit::noise_generators::CartesianConstraintsSampling,stomp_moveit::noise_generators::StompNoiseGenerator);
+
+static const std::vector<double> ACC_MATRIX_DIAGONAL_VALUES = {-1.0/12.0, 16.0/12.0, -30.0/12.0, 16.0/12.0, -1.0/12.0};
+static const std::vector<int> ACC_MATRIX_DIAGONAL_INDICES = {-2, -1, 0 ,1, 2};
+static const int CARTESIAN_DOF_SIZE = 6;
+static const double DEFAULT_IK_POS_TOLERANCE = 0.001;
+static const double DEFAULT_IK_ROT_TOLERANCE = 0.01;
+
+
+namespace stomp_moveit
+{
+namespace noise_generators
+{
+
+CartesianConstraintsSampling::CartesianConstraintsSampling():
+  name_("CartesianConstraintsSampling"),
+  goal_rand_generator_(new RandomGenerator(RGNType(),boost::uniform_real<>(-1,1)))
+{
+
+}
+
+CartesianConstraintsSampling::~CartesianConstraintsSampling()
+{
+
+}
+
+bool CartesianConstraintsSampling::initialize(moveit::core::RobotModelConstPtr robot_model_ptr,
+                        const std::string& group_name,const XmlRpc::XmlRpcValue& config)
+{
+  using namespace moveit::core;
+
+  // robot model details
+  group_ = group_name;
+  robot_model_ = robot_model_ptr;
+  const JointModelGroup* joint_group = robot_model_ptr->getJointModelGroup(group_name);
+  if(!joint_group)
+  {
+    ROS_ERROR("Invalid joint group %s",group_name.c_str());
+    return false;
+  }
+
+  // kinematics
+  ik_solver_.reset(new stomp_moveit::utils::kinematics::IKSolver(robot_model_ptr,group_name));
+
+  // trajectory noise generation
+  stddev_.resize(CARTESIAN_DOF_SIZE);
+
+  // creating default cartesian tolerance for IK solver
+  tool_goal_tolerance_.resize(CARTESIAN_DOF_SIZE);
+  double ptol = DEFAULT_IK_POS_TOLERANCE;
+  double rtol = DEFAULT_IK_ROT_TOLERANCE;
+  tool_goal_tolerance_ << ptol, ptol, ptol, rtol, rtol, rtol;
```

Comment:
I would use std::array and statically initialize this in header.

Reply:
tool_goal_tolerance_ is type Eigen::VectorXd and it is use as input argument IK solver

Reply:
stddev_ -> std::array